pkg/fanal/analyzer/language/c/conan/conan.go

```diff
@@ -1,42 +1,152 @@
+	return conanLockAnalyzer{
+		logger: log.WithPrefix("conan"),
+		parser: conan.NewParser(),
+	}, nil
+}
 
-func (a conanLockAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
-	p := conan.NewParser()
-	res, err := language.Analyze(types.Conan, input.FilePath, input.Content, p)
+func (a conanLockAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
+	required := func(filePath string, d fs.DirEntry) bool {
+		return a.Required(filePath, nil)
+	}
+
+	licenses, err := licensesFromCache()
 	if err != nil {
-		return nil, xerrors.Errorf("%s parse error: %w", input.FilePath, err)
+		a.logger.Debug("Unable to parse cache directory to obtain licenses", log.Err(err))
+	}
+
+	var apps []types.Application
+	if err = fsutils.WalkDir(input.FS, ".", required, func(filePath string, _ fs.DirEntry, r io.Reader) error {
+		app, err := language.Parse(types.Conan, filePath, r, a.parser)
+		if err != nil {
+			return xerrors.Errorf("%s parse error: %w", filePath, err)
+		}
+
+		if app == nil {
+			return nil
+		}
+
+		// Fill licenses
+		for i, lib := range app.Libraries {
+			if license, ok := licenses[lib.Name]; ok {
+				app.Libraries[i].Licenses = []string{
+					license,
+				}
+			}
+		}
+
+		sort.Sort(app.Libraries)
+		apps = append(apps, *app)
+		return nil
+	}); err != nil {
+		return nil, xerrors.Errorf("unable to parse conan lock file: %w", err)
+	}
+
+	return &analyzer.AnalysisResult{
+		Applications: apps,
+	}, nil
+}
+
+func licensesFromCache() (map[string]string, error) {
+	required := func(filePath string, d fs.DirEntry) bool {
+		return filepath.Base(filePath) == "conanfile.py"
+	}
+
+	// cf. https://docs.conan.io/1/mastering/custom_cache.html
+	cacheDir := os.Getenv("CONAN_USER_HOME")
+	if cacheDir == "" {
+		cacheDir, _ = os.UserHomeDir()
+	}
+	cacheDir = path.Join(cacheDir, ".conan", "data")
+
+	if !fsutils.DirExists(cacheDir) {
+		return nil, xerrors.Errorf("the Conan cache directory (%s) was not found.", cacheDir)
+	}
+
+	licenses := make(map[string]string)
+	if err := fsutils.WalkDir(os.DirFS(cacheDir), ".", required, func(filePath string, _ fs.DirEntry, r io.Reader) error {
+		scanner := bufio.NewScanner(r)
+		var name, license string
+		for scanner.Scan() {
+			line := strings.TrimSpace(scanner.Text())
+
+			if strings.HasPrefix(line, "name") { // cf. https://docs.conan.io/1/reference/conanfile/attributes.html#name
+				// remove spaces before and after `=` (if used). e.g. `name = "openssl"` -> `name="openssl"`
+				name = strings.ReplaceAll(line, " ", "")
+				// trim extra characters - e.g. `name="openssl"` -> `openssl`
+				name = strings.TrimSuffix(strings.TrimPrefix(name, `name="`), `"`)
+				// Check that the license is already found
+				if license != "" {
+					break
+				}
+			} else if strings.HasPrefix(line, "license") { // cf. https://docs.conan.io/1/reference/conanfile/attributes.html#license
```

Comment:
It should be more strict. Otherwise, it matches other than the license line. That's why I suggested regexp this time.
https://github.com/conan-io/conan-center-index/blob/55703762e2ca0560fd44d9008e00e61a0a64b23c/recipes/zlib/all/conanfile.py#L90

Reply:
it looks like we can't avoid using regexp...

Reply:
If possible, I don't want to use regexp. I'm still considering a way to parse the license line without regexp, but...

Reply:
take a look this -https://go.dev/play/p/FkXPq_6nS40

Reply:
Yes, it's better.